File: tests/indexing/parsing/lang/fixtures/java/Lambdas.java
package pipeline;

import java.util.List;
import java.util.function.Function;
import java.util.stream.Collectors;

public class Processor {
    private Runnable onComplete;

    public List<Integer> transform(List<Integer> values) {
        Function<Integer, Integer> doubler = x -> x * 2;
        return values.stream()
                .filter(v -> v > 0)
                .map(v -> {
                    int scaled = doubler.apply(v);
                    return scaled + 1;
                })
                .collect(Collectors.toList());
    }

    public void configure() {
        this.onComplete = () -> {
            System.out.println("done");
        };
    }
}

File: tests/indexing/parsing/lang/fixtures/kotlin/Lambdas.kt
package pipeline

class Processor {
    private var onComplete: (() -> Unit)? = null

    fun transform(values: List<Int>): List<Int> {
        val doubler = fun(x: Int): Int { return x * 2 }
        return values
            .filter { it > 0 }
            .map(fun(v: Int): Int {
                val scaled = doubler(v)
                return scaled + 1
            })
    }

    fun configure() {
        this.onComplete = fun() {
            println("done")
        }
    }
}
